File: leetcode/2221. 数组的三角和.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int triangularSum(vector<int>& nums) {
        queue<int> q;
        for (int num : nums) {
            q.push(num);
        }
        while (q.size() > 1) {
            int n = q.size();
            for (int i = 0; i < n - 1; ++i) {
                int a = q.front();
                q.pop();
                int b = q.front();
                q.push((a + b) % 10);
                if (i == n - 2) {
                    q.pop();
                }
            }
        }
        return q.front();
    }
};

File: leetcode/2221. 数组的三角和.rs
impl Solution {
    pub fn triangular_sum(nums: Vec<i32>) -> i32 {
        use std::collections as queue;
        let mut q: queue::VecDeque<i32> = nums.into();
        while q.len() > 1 {
            let n = q.len();
            for i in 0..n - 1 {
                let a = q.pop_front().unwrap();
                let b = q.front().unwrap();
                q.push_back((a + b) % 10);
                if i == n - 2 {
                    q.pop_front();
                }
            }
        }
        q.front().cloned().unwrap()
    }
}